vendor/k8s.io/kubernetes/pkg/controller/statefulset/stateful_set.go

```diff
@@ -450,7 +450,11 @@ func (ssc *StatefulSetController) sync(key string) error {
 // syncStatefulSet syncs a tuple of (statefulset, []*v1.Pod).
 func (ssc *StatefulSetController) syncStatefulSet(set *apps.StatefulSet, pods []*v1.Pod) error {
 	glog.V(4).Infof("Syncing StatefulSet %v/%v with %d pods", set.Namespace, set.Name, len(pods))
-	if err := ssc.control.UpdateStatefulSet(set, pods); err != nil {
+	setCopy, err := api.Scheme.DeepCopy(set)
```

Comment:
does this looks suspicious? https://github.com/mfojtik/kubernetes/blob/35d6cf086c38f4416f72c91c8af138e8eb3e4090/pkg/controller/statefulset/stateful_set_utils.go#L303